dist/pythonlibs/riotctrl_shell/cord_ep.py

```diff
@@ -0,0 +1,142 @@
+        >>> res = parser.parse("CoAP RD connection status:\\n"
+        ...             "RD address:"
+        ...             " coaps://[fe80::48d5:9eff:fe98:6b74%iface0]:5684\\n"
+        ...             "   ep name: RIOT-760D2323760D2323\\n"
+        ...             "  lifetime: 60s\\n"
+        ...             "    reg if: /resourcedirectory\\n"
+        ...             "  location: /reg/1/")
+        >>> res['rdaddr']
+        'coaps://[fe80::48d5:9eff:fe98:6b74%iface0]:5684'
+        """
+        res = {}
+        for line in cmd_output.splitlines():
+            for key, comp in self.comps.items():
+                m = comp.search(line)
+                if m is not None:
+                    try:
+                        res[key] = int(m.group(key))
+                    except ValueError:
+                        res[key] = m.group(key)
+        return res if bool(res) else None
+
+
+class CordEpDiscoverParser(ShellInteractionParser):
+
+    def __init__(self):
+        self.comp = re.compile(r"the registration interface is"
+                               r"\s+'(?P<regif>\S+)'$")
+
+    def parse(self, cmd_output):
+        """
+        Parses output of CordEp::cord_ep_discover()
+
+        >>> parser = CordEpDiscoverParser()
+        >>> parser.parse(
+        ...     "the registration interface is '/resourcedirectory'")
+        '/resourcedirectory'
+        """
+        for line in cmd_output.splitlines():
+            m = self.comp.search(line)
+            if m is not None:
+                return m.group("regif")
+        return None
+
+
+# ==== ShellInteractions ====
+
+
+class CordEp(ShellInteraction):
+    REGISTER = "register"
+    DISCOVER = "discover"
+    UPDATE = "update"
+    REMOVE = "remove"
+    INFO = "info"
+
+    @staticmethod
+    def _make_addr(addr, port=None):
+        addr = f"[{addr}]"
+        if port is not None:
+            addr += ":{}".format(port)
+        return addr
+
+    @ShellInteraction.check_term
+    def cord_ep_cmd(self, cmd, args=None, timeout=-1, async_=False):
+        cmd_str = "cord_ep {cmd}".format(cmd=cmd)
+        if args is not None:
+            cmd_str += " {args}".format(args=" ".join(str(a) for a in args))
+        res = self.cmd(cmd_str, timeout=timeout, async_=async_)
+        if ("error:" in res) or ("usage:" in res):
+            raise RuntimeError(res)
+        return res
+
+    def cord_ep_info(self, timeout=-1, async_=False):
+        return self.cord_ep_cmd(self.INFO, None, timeout, async_)
+
+    def cord_ep_register(self, addr, port=None, regif=None,
+                         timeout=-1, async_=False):
+        args = [self._make_addr(addr, port)]
+        if regif is not None:
+            args.append(regif)
+        return self.cord_ep_cmd(self.REGISTER, args, timeout, async_)
+
+    def cord_ep_discover(self, addr, port=None, timeout=-1, async_=False):
+        args = [self._make_addr(addr, port)]
+        return self.cord_ep_cmd(self.DISCOVER, args, timeout, async_)
```

Comment:
```suggestion
    def cord_ep_discover(self, uri, timeout=-1, async_=False):
        return self.cord_ep_cmd(self.DISCOVER, (uri,), timeout, async_)
```

Reply:
is there any difference putting `uri` in a tuple `(uri,)` instead of a list `[uri]` here?

Reply:
No, but since you don't change the `args` from here on out, I thought it better to use an immutable object here (tuples are immutable, lists are not)